core/src/main/java/org/apache/cxf/endpoint/ClientImpl.java

```diff
@@ -1083,12 +1087,29 @@ public void clear() {
         }
     }
 
-
     public void setExecutor(Executor executor) {
         if (!SynchronousExecutor.isA(executor)) {
             this.executor = executor;
         }
     }
 
+    /**
+     * Returns the current thread name. As this value will be used as a key in a {@link WeakHashMap},
+     * we should use the String constructor for avoiding scenarios where strong references are kept
+     * to the string names causing that the map entries can't be garbage collected.
+     * @return
+     */
+    private String getThreadName() {
```

Comment:
you probably want to use the thread id and not the name which doesn't have to be unique?

Reply:
Also, you're not going to want to `new` the string each time, for performance reasons.  Can the methods that call this reduce the number of times they call it?